### PR TITLE
Bump org.eclipse.equinox.p2.core to 2.12.0

### PR DESCRIPTION
**Pull request by merks**
Fixes https://github.com/eclipse-equinox/p2/issues/514

### CODE REVIEW

**Review by iloveeclipse — CHANGES_REQUESTED** [review]
Looking on dependencies of the new API, following bundles should enforce new version:
- org.eclipse.equinox.p2.metadata.repository
- org.eclipse.equinox.p2.ui.sdk.scheduler

**Comment by merks** [comment]
I won't have time for further changes after my 13 hour day.  If you feel they are necessary, you or someone else will need to make such changes.  Sorry.

**Review by iloveeclipse — APPROVED** [review]

File: bundles/org.eclipse.equinox.p2.core/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -74,7 +74,7 @@ Service-Component:
 Import-Package: org.bouncycastle.bcpg;version="1.65.0",
  org.bouncycastle.openpgp;version="1.65.0",
  org.eclipse.core.runtime;common=split;version="[3.5.0,4.0.0)",
- org.eclipse.equinox.p2.core;version="[2.8.0,2.9.0)";resolution:=optional,
+ org.eclipse.equinox.p2.core;version="[2.12.0,2.13.0]";resolution:=optional,
```
Comment (iloveeclipse) [inline bundles/org.eclipse.equinox.p2.core/META-INF/MANIFEST.MF:77]: `org.eclipse.equinox.p2.core;version="[2.12.0,2.13.0]"` ???

Why do we import same package we provide? Is this something that can be provided by others???

**Comment by iloveeclipse** [comment]
Interesting (and unrelated) error on Mac

```
Failed to execute goal org.eclipse.tycho:tycho-apitools-plugin:4.0.8-SNAPSHOT:verify (verify) on project org.eclipse.equinox.p2.discovery: 
Execute ApiApplication failed: InvocationTargetException: 
Cannot load from object array because "this.this$0.fIndexedDetectors" is null -> [Help 1]
```
Shouldn't be the problem caused by this PR.

**Review by iloveeclipse — APPROVED** [review]

**Comment by iloveeclipse** [comment]
Something is wrong with API tooling:
```
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-apitools-plugin:4.0.8-SNAPSHOT:verify (verify) on project org.eclipse.equinox.p2.reconciler.dropins: Execute ApiApplication failed: InvocationTargetException: Error reading manifest for /home/jenkins/agent/workspace/p2_PR-516/.m2/repository/p2/osgi/bundle/org.sat4j.pb.source/2.3.6.v20201214/org.sat4j.pb.source-2.3.6.v20201214.jar: zip file is empty
```

File: bundles/org.eclipse.equinox.p2.core/META-INF/MANIFEST.MF (hunk 1 above)
Comment (HannesWell) [reply bundles/org.eclipse.equinox.p2.core/META-INF/MANIFEST.MF]: > Why do we import same package we provide? Is this something that can be provided by others???

Importing exported packages is considered good OSGi practice by many:
https://felix.apache.org/documentation/tutorials-examples-and-presentations/apache-felix-osgi-faq.html#import-own-exports
I have not yet seen the necessity of that for Eclipse, but I have also not thought about it in great depth.

**Pull request merged**